test/unittest/readertest.cpp

```diff
@@ -190,7 +191,7 @@ static void TestParseDouble() {
         internal::Double e(x), a(h.actual_); \
         EXPECT_EQ(e.Sign(), a.Sign()); \
         if (fullPrecision) { \
-            EXPECT_EQ(x, h.actual_); \
+            EXPECT_NEAR(x, h.actual_, 0.0); \

```

Comment:
Sadly the warnings still appear with suppression. I think it is because the headers are included firstly than the pragmas. I try to change it with binary representation comparison.